### DISCUSSION TITLE
Developement of Solid Beam element formulation

### DISCUSSION THREAD

**Discussion by abdullah2146** [idaholab/moose#21049] [Q&A General] [answered]
Hi,

I want to develop the three-dimensional Solid beam formulation which is developed in this paper.

https://www.sciencedirect.com/science/article/pii/S0045782513001618 

Has the MOOSE already developed this formulation? If not, can i develop this formulation on MOOSE? 

This formulation have different matrics which need to define like  

![image](https://user-images.githubusercontent.com/104014684/169144642-2258e7b7-8a8a-49c6-9c43-d6445412b4c2.png)

Can you please answer this question?

Thanks,

**Comment by GiudGiud** [comment]
@jiangwen84 @dschwen

**Comment by recuero** [comment, marked as answer]
No, MOOSE doesn't have a solid beam formulation. The existing beam formulation makes use of the typical centerline plus cross section rotation approach (i.e. Timoshenko) with torsion (see https://mooseframework.inl.gov/modules/tensor_mechanics/beam_vandv.html).

**Reply by abdullah2146** [reply]
Thanks for your reply. As it's not developed on MOOSE, Can I develop this Solid beam formulation on my own?

**Reply by GiudGiud** [reply]
Certainly. 
And contributions are always welcome if this goes well.

**Reply by abdullah2146** [reply]
@GiudGiud Can you share any C++ files or guidance how we can develop formulation in MOOSE?

**Reply by GiudGiud** [reply]
If you are a beginner in MOOSE you should follow the tutorials here first
https://mooseframework.inl.gov/index.html
If not you should consult the tensor mechanics documentation to see how various formulations were implemented
https://mooseframework.inl.gov/modules/tensor_mechanics/index.html
